File: src/main/java/field/movement/fragment/StatChangeMoveFragment.java
package field.movement.fragment;

import field.object.life.FieldLife;
import util.packet.PacketReader;
import util.packet.PacketWriter;

public class StatChangeMoveFragment extends AbstractMovementFragment {

    boolean stat;

    public StatChangeMoveFragment(byte movePathAttribute, PacketReader packetReader) {
        super(movePathAttribute, packetReader);
    }

    @Override
    public void apply(FieldLife life) {

    }

    @Override
    public void decodeData(PacketReader packet) {
        stat = packet.readBool();
    }

    @Override
    public void encodeData(PacketWriter packet) {
        packet.writeBool(stat);
    }
}

File: src/main/java/field/movement/fragment/StatChangeMoveFragment.kt
package field.movement.fragment

import field.obj.life.FieldLife
import util.packet.PacketReader
import util.packet.PacketWriter

class StatChangeMoveFragment(movePathAttribute: Byte, packetReader: PacketReader) : AbstractMovementFragment(movePathAttribute, packetReader) {

    var stat = false
    override fun apply(life: FieldLife) {}

    override fun decodeData(packet: PacketReader) {
        stat = packet.readBool()
    }

    override fun encodeData(packet: PacketWriter) {
        packet.writeBool(stat)
    }
}